apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/Keys.cs

```diff
@@ -0,0 +1,319 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Protobuf.WellKnownTypes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Cloud.Spanner.Data
+{
+    /// <summary>
+    /// Defines a set of keys to select from a table or index.
+    /// </summary>
+    public class KeySet
```

Comment:
`KeySet` is currently only used for `Read` operations, but these could also be used for `Delete` mutation commands. The current `Delete` command implementation only accepts one `SpannerParameterCollection` as an input parameter, and that collection is translated to exactly one key, which means that it is currently only possible to delete one row at a time using mutations. We could also enable bulk deletes through mutations by adding an overload for `CreateDeleteCommand` that accepts a `KeySet` as an input parameter.

Reply:
+1 to this, as long as we continue to support the Parameters for Deletes, for anyone out there using it.